src/js/MazeEntity.js

```diff
@@ -2,6 +2,18 @@ class MazeEntity {
     constructor(maze, row, col) {
         Object.assign(this, {maze, row, col});
     }
+    getNeighboringEntitiesDict(distance){
+      const data = this.maze.data, row = this.row, col = this.col;
+      return {
+        'left': col > (distance-1)? data[row][col-distance] : null,
+        'right': col < data[0].length - distance ? data[row][col+distance] : null,
+        'top':  row > (distance-1) ? data[row-distance][col] : null,
+        'bottom': row < data.length - distance ? data[row+distance][col] : null
+      }
+    }
+    getNeighboringEntitiesList(distance){
```

Comment:
how about `listNeighborsAtDist()`, using `list` as a verb? idk. also, do we def use this enough places for it to make sense as a separate function? (also also, see my note above about filtering out the `null`s.)

Reply:
Ahhh yeah, filtering the nulls is totally better, will do